### ISSUE TITLE
Outfile problems with --include option

### ISSUE THREAD

**Issue by kreddkrikk** [inotify-tools/inotify-tools#168]
I am using the latest release (3.22.6.0) and am having issues with the `--include` option. If I create a subdirectory in a watched directory, file creation events don't get logged if a file that matches the `--include` regex pattern was created in that new subdirectory (unless the subdirectory _also_ matches the `--include` regex pattern). I also provide the `-r` option for recursive watching. This happens in both daemon and monitor mode:
```sh
inotifywait -r -d -o /path/to/outfilename.log -e create --include "\.bin$" /path/to/dirtowatch
mkdir /path/to/dirtowatch/newdir # Does not get logged
mkdir /path/to/dirtowatch/newdir.bin # Does get logged
touch /path/to/dirtowatch/newdir/file.bin # Does NOT get logged, even though file.bin matches --include regex pattern
touch /path/to/dirtowatch/newdir.bin/file.bin # Does get logged, maybe because newdir.bin was logged?
```
Is this because file/directory creation events that don't match the `--include` pattern are completely discarded, hence any new files created in new unmatched subdirectories are not logged?

Also, when I run in daemon mode (`-d`) with an outfile specified (`-o`), it does not automatically create the file if it doesn't exist. There are no issues when I switch to monitor mode (`-m`). Here is an example command:
```sh
inotifywait -r -d -o /path/to/outfilename.log -e create /path/to/dirtowatch
```
This error appears when I run the above command:
```sh
No such file or directory: /path/to/outfilename.log
```
If I create /path/to/outfilename.log before starting inotifywait, this error doesn't appear. Is this normal behavior?

**Comment by ericcurtin** [comment]
> If I create /path/to/outfilename.log before starting inotifywait, this error doesn't appear. Is this normal behavior?

This normal behaviour, although I am open to enhancing it

**Comment by Obeyed** [comment]
Regarding the fact that `--include` and `--includei` ignores files when directories are created while watching the parent directory—should we open a separate issue specifically for this, or what is the preference in this case?

I have verified that `inotifywait` ignores relevant files with the **--include(i)** param when directories are created while the parent directory is being watched. If we start `inotifywait` while the subdirectories already exist then it catches the files as expected.

I have `inotifywait` version **3.22.1.0** installed on Ubuntu 22.04.

**Comment by ericcurtin** [comment]
I'm happy with either way you want to track it.

**Comment by ericcurtin** [comment]
@f1d094 is a completely unfunded project, if behaviour is desired to be change, somebody must write the code, somebody must make the contribution, somebody must open the pull request.

We are 100% a community project.
